### DISCUSSION TITLE
Use msconvert in Linux or Mac - Miao Yu | 于淼

### DISCUSSION THREAD

**Discussion by yufree** [yufree/yufree.cn#372] [General]
Use msconvert in Linux or Mac 
 https://yufree.cn/en/2019/10/15/use-msconvert-in-linux-or-mac/

**Comment by yufree** [comment]
`Originally posted by 申祖涛 on 2021-01-20 03:23:32
        `
Great!

**Comment by yufree** [comment]
`Originally posted by slum on 2022-08-31 12:37:02
        `
我想问一下，我按您的代码转换数据，程序是能够运行，看到&quot;writing output
file: &quot;,但是却始终没有发现转化后的.mzML文件。

**Reply by yufree** [reply]
`Originally posted on 2022-09-01 00:40:13
        `
如果默认的话原有mzML文件会跟原文件出现在同一位置，另外就是 -v
后面的文件位置映射到了镜像中的 data
文件夹下，所以输出也一定要在data文件夹下才会映射回原文件位置。

**Reply by yufree** [reply]
`Originally posted by slum on 2022-09-01 01:33:49
        `
谢谢，现在可以了！

**Comment by Ran485** [comment]
"processing file: Exp_75min_15cm_500ng_20230914_F1_R1.RAW
[RawFileImpl::getInstrumentMethods()] Access to the path 'C:\users\root\Temp\tmpc7da.tmp' is denied.
Error processing file Exp_75min_15cm_500ng_20230914_F1_R1.RAW" 请教一下，这种错误该怎么解决呀，好像就是只有转“.raw”格式的文件的时候才会出现这个问题

**Reply by yufree** [reply]
Thermo的还是用他们自己出的ThermoRawFileParser来转换吧，msconvert对型号比较新的仪器转换存在问题。

**Comment by CXMMMMMM** [comment]
可以请问一下为什么 在Mac上开terminal运行, 一只都是permission denied呀
所有目录里的文件都是drwxrwxrwx了呀

**Reply by CXMMMMMM** [reply]
啊 对不 是另一个问题, 我要转化的是Waters的raw文件, 在M4 的 Mac OS上, 他总是提示匹配不到文件,
是我平台的问题吗
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
[msconvert] no files found matching "./Test_MSconvert/140324_008.RAW"

**Reply by yufree** [reply]
是平台问题arm构架CPU没法用 另外你路径可能也有问题

**Comment by Wenti1996** [comment]
你们好，我刚刚开始学习代谢组学相关的各种软件工具使用，若是各位大神有时间的话能留下一个email地址吗？我希望能和你们交流学习，特别是关于MSConvert在MacOS(M4)上的运行

**Reply by CXMMMMMM** [reply]
直接找公司要mzML最快

**Reply by Wenti1996** [reply]
不好意思， 没有太懂找公司要的意思
